[C#] Add inheritance exercise (#1424)
exercism/v3: languages/csharp/exercises/concept/inheritance/.docs/after.md

> In C#, a _class_ hierarchy can be defined using _inheritance_, which allows a derived class (`Car`) to inherit the behavior and data of its parent (general) class (`Vehicle`). If no parent is specified, the class inherits from the `object` class. To prevent a class being inherited, add the [`sealed` modifier][sealed-classes].

Reviewer: ```suggestion
In C#, a _class_ hierarchy can be defined using _inheritance_. This allows a derived class (`Car`) to inherit the behavior and data of its parent class (`Vehicle`). If no parent is specified, the class inherits from the `object` class. To prevent a class being inherited, add the [`sealed` modifier][sealed-classes].
```
